ariadne/cardano/test/backend/Test/Spec/Fixture.hs

```diff
@@ -39,13 +43,27 @@ genSpendingPassword :: PropertyM IO PassPhrase
 genSpendingPassword =
     pick (frequency [(20, pure emptyPassphrase), (80, arbitrary)])
 
+genSpendingPasswords :: Int -> PropertyM IO [PassPhrase]
+genSpendingPasswords n = pick $ vector n
```

Comment:
So it uses `instance Arbitrary PassPhrase`. Logic there differs from `genSpendingPassword`.
It looks weird to me that `genSpendingPassword` and `genSpendingPasswords` use different logic to generate password.
I would expect this function to be defined as `genSpendingPasswords n = replicateM n genSpendingPassword`.

Reply:
`genSpendingPassword` for some reasons generate 20% of empty passphrases, but I need to have a list of unique. I even filter the list of random values. So - the cause of using different logic is that I don't want to generate 2 empty passwords and filter them every time. 

I will ask @Pastafarianist if I can remove that 20% of empty passphrases.

Reply:
I need unique passphrases because with the same passphrases I get the same requests and error when I try to create a wallet with the same ID.

Reply:
I don't get it.
1. `instance Arbitrary PassPhrase` is defined as follows:
  ```haskell
  instance Arbitrary PassPhrase where
      arbitrary = oneof [
          pure mempty,
          ByteArray.pack <$> vector 32
          ]
  ```
  So it generates empty passphrase with 50% probability, which is even greater than probability of generating empty passphrase in `genSpendingPassword`.
2.
> I need unique passphrases because with the same passphrases I get the same requests and error when I try to create a wallet with the same ID.

Why do you get the same requests? `genNewWalletRq` picks arbitrary `eskNoPass`. It doesn't depend on passphrase at all. If you get the same requests, it means that `eskNoPass      <- pick arbitrary` generates the same wallet IDs, which is unrelated to generation of passphrases.

Reply:
1. Oh! thanks! I thought that I get errors because random values can be duplicated and that `Arbitrary` instance should return really random value by default. 
Anyway, I'll add function to get a really random list of passphrases (I don't want to change the instance because I believe that there were reasons to get 50% of empty passphrases). 
(I thought "why I get so many duplications ??")

And yes, you are right - I'll use `genSpendingPassword` function.

Reply:
> It doesn't depend on passphrase at all. 

It is strange because with unique passphrases it actually generates different wallets. 
If i remove `OrdNub` from ```requests <- mapM genNewWalletRq $ ordNub passwds``` it generates sometimes duplicated requests (and duplicated wallets as well).